src/stdune/file.mli

```diff
@@ -0,0 +1,12 @@
+type t =
```

Comment:
If we move this module to a more  central place, its name needs to identify more universally what this module is about. `File` doesn't seem like the right name to me.

Reply:
Yeah the name isn't great. Happy to take suggestions

Reply:
I don't have a better suggestions apart from something long and descriptive such as `Inode_and_device_number`. Given the size of this module, it makes me think that we should keep it as a sub-module for now.

Reply:
Indeed. I moved it back.